packages/ra-core/src/controller/RecordContext.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { createContext, useContext, ReactNode } from 'react';
+import pick from 'lodash/pick';
+import { Record } from '../types';
+
+/**
+ * Context to store the result of the useRecord() hook.
+ *
+ * Use the useRecordContext() hook to read the context. That's what the Edit and Show components do in react-admn.
+ *
+ * @example
+ *
+ * import { useEditController, EditContext } from 'ra-core';
+ *
+ * const Edit = props => {
+ *     const controllerProps = useEditController(props);
+ *     return (
+ *         <RecordContextProvider value={controllerProps}>
+ *             ...
+ *         </RecordContextProvider>
+ *     );
+ * };
+ */
+export const RecordContext = createContext<Record | Omit<Record, 'id'>>(
+    undefined
+);
+
+export const RecordContextProvider = ({
+    children,
+    value,
+}: RecordContextOptions) => (
+    <RecordContext.Provider value={value}>{children}</RecordContext.Provider>
+);
+RecordContext.displayName = 'RecordContext';
+
+export const usePickRecordContext = <
```

Comment:
I'd call it `pickRecord` because:

- it's not a hook
- it can pick record from anything, not just a context

Same for other pickers by the way

Reply:
Should I do the same for all pickers which were already there for the ListContextProvider ?

Reply:
yep

Reply:
Not possible as they use hooks